src/core/account/utils/address.ts

```diff
@@ -0,0 +1,39 @@
+import { sha3_256 } from "@noble/hashes/sha3";
+import { AccountAddress } from "../../accountAddress";
+import { DeriveScheme } from "../../../types";
+
+/**
+ * Creates an object address from creator address and seed
+ *
+ * @param creatorAddress The object creator account address
+ * @param seed The seed in either Uint8Array | string type
+ *
+ * @returns The object address
+ */
+export const createObjectAddress = (creatorAddress: AccountAddress, seed: Uint8Array | string): string => {
+  const creatorBytes = creatorAddress.bcsToBytes();
+
+  const seedBytes = typeof seed === "string" ? Buffer.from(seed, "utf8") : seed;
+
+  const bytes = new Uint8Array([...creatorBytes, ...seedBytes, DeriveScheme.DeriveObjectAddressFromSeed]);
+
+  return new AccountAddress(sha3_256(bytes)).toString();
+};
```

Comment:
Why not just return the AccountAddress instead of the string?

Reply:
honestly, no strong opinion here - this should be based on end user usage and convenience. Might be better to just return the  AccountAddress and let the end user to figure our what format they want it in

Reply:
updated